docs/configuration/uds-operator.md

```diff
@@ -207,6 +207,48 @@ spec:
 
 This configuration does not create a secret in the cluster and instead tells the UDS Operator to create a public client (one that requires no auth secret) that enables the `oauth2.device.authorization.grant.enabled` flow and disables the standard redirect auth flow.  Because this creates a public client configuration that deviates from this is limited - if your application requires both the Device Authorization Grant and the standard flow this is currently not supported without creating two separate clients.
 
+### Creating a UDS Package with a Service Account Roles client
+
+Some applications may need to access resources / obtain OAuth tokens on behalf of *themselves* vice users. This may be needed to allow API access to Authservice protected applications (outside of a web browser). This is commonly used in machine-to-machine authentication for automated processes. This type of grant in OAuth 2.0 is known as the [Client Credentials Grant](https://oauth.net/2/grant-types/client-credentials/) and is supported in a UDS Package with the following configuration:
+
+```yaml
+apiVersion: uds.dev/v1alpha1
+kind: Package
+metadata:
+  name: client-cred
+  namespace: argo
+spec:
+  sso:
+    - name: httpbin-api-client
+      clientId: httpbin-api-client
+      standardFlowEnabled: false
+      serviceAccountsEnabled: true
+
+      # By default, Keycloak will not set the audience `aud` claim for service account access token JWTs.
+      # You can optionally add a protocolMapper to set the audience.
+      # If you map the audience to the same client used for authservice, you can enable access to authservice protected apps with a service account JWT.
+      protocolMappers:
+        - name: audience
+          protocol: "openid-connect"
+          protocolMapper: "oidc-audience-mapper"
+          config:
+            included.client.audience: "uds-core-httpbin" # Set this to match the app's authservice client id
+            access.token.claim: "true"
+            introspection.token.claim: "true"
+            id.token.claim: "false"
+            lightweight.claim: "false"
+            userinfo.token.claim: "false"
+
+      # Optionally, store the client secret as a Kubernetes secret
+      secretName: httpbin-api-client
+      secretTemplate:
+        clientId: "clientField(clientId)"
+        clientSecret: "clientField(secret)"
```

Comment:
Follow up to the previous thread and suggestion: https://github.com/defenseunicorns/uds-core/pull/852#discussion_r1792162389

Two paths I think we could take here:
1. Just delete this since we capture the SSO secret templating separately [here](https://github.com/defenseunicorns/uds-core/blob/main/docs/configuration/uds-operator.md#example-uds-package-cr-with-sso-templating). I don't _think_ we need to call it out here again?
2. Update the comment to be a bit more accurate:
```suggestion

      # Optionally, provide a custom secret name and template for the secret contents
      secretName: httpbin-api-client
      secretTemplate:
        clientId: "clientField(clientId)"
        clientSecret: "clientField(secret)"
```

Reply:
If it's captured elsewhere it may make more sense to delete it so we're not crossing the streams too much?